messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -272,6 +285,7 @@ protected void processingLoop(Segment segment) {
                     && canClaimSegment(segment.getSegmentId())) {
                 try {
                     eventStream = ensureEventStreamOpened(eventStream, segment);
+                    processScopeEvents();
```

Comment:
`processScopeEvents` is called after `ensureEventStreamOpened` to ensure this `TrackingEventProcessor` actually has a claim on a token, I guess? 
If so, is that even important when it comes to deadlines / scoped events?

Reply:
Actually not, it's important to have scope events processed before "regular" events due to their urgency. However, you have a point here since we don't have to open a stream before processing scope events.

On a completely different subject, this is where I question whether this whole changeset makes sense since scoped events (deadlines) will be executed with a delay. With the previous approach, we didn't have that issue...

Reply:
Exactly true. Had some discussion with Allard on it as well, think a meeting on this is justified.